index.js

```diff
@@ -99,238 +105,323 @@ const defaultProps = {
+    componentDidUpdate(prevProps) {
         let newState = {};
         let doUpdate = false;
         if (prevProps.initValue !== this.props.initValue) {
-            newState.selected = this.props.initValue;
+            newState.selected = [this.props.initValue];
             doUpdate = true;
         }
         if (prevProps.visible !== this.props.visible) {
             newState.modalVisible = this.props.visible;
             doUpdate = true;
         }
-        if(prevProps.selectedKey !== this.props.selectedKey){
-            let selectedItem = this.validateSelectedKey(this.props.selectedKey);
-            newState.selected = selectedItem.label;
-            newState.changedItem = selectedItem.key;
+        if (
+            JSON.stringify(prevProps.initSelectedKeys) !==
+      JSON.stringify(this.props.initSelectedKeys)
+        ) {
+            let validatedKeys = this.props.initSelectedKeys.reduce((acc, val) => {
+                const { key } = this.validateSelectedKey(val);
+                key && acc.push(key);
+            }, []);
+            newState.selected = validatedKeys;
             doUpdate = true;
         }
         if (doUpdate) {
             this.setState(newState);
         }
     }
 
-    validateSelectedKey = (key) => {
-        let selectedItem = this.props.data.filter((item) => this.props.keyExtractor(item) === key);
-        let selectedLabel = selectedItem.length > 0 ? this.props.labelExtractor(selectedItem[0]) : this.props.initValue;
-        let selectedKey = selectedItem.length > 0 ? key : undefined;
-        return {label: selectedLabel, key: selectedKey}
-    }
+  validateSelectedKey = key => {
+      let selectedItem = this.props.data.filter(
+          item => this.props.keyExtractor(item) === key
+      );
+      let selectedLabel =
+      selectedItem.length > 0
+          ? this.props.labelExtractor(selectedItem[0])
+          : this.props.initValue;
+      let selectedKey = selectedItem.length > 0 ? key : undefined;
+      return { label: selectedLabel, key: selectedKey };
+  };
 
-    onChange = (item) => {
-        if (Platform.OS === 'android' || (Modal.propTypes !== undefined && !Modal.propTypes.onDismiss)) { // don't know if this will work for previous version, please check!
-            // RN >= 0.50 on iOS comes with the onDismiss prop for Modal which solves RN issue #10471
-            this.props.onChange(item);
-        }
-        this.setState({ selected: this.props.labelExtractor(item), changedItem: item }, () => {
-          if (this.props.closeOnChange)
-            this.close();
-        });
-    }
+  onChange = (item, checked = false) => {
+      let { selected } = this.state;
+      const itemKey = this.props.keyExtractor(item);
+      if (this.props.multiple)
+          checked
+              ? selected.push(itemKey)
+              : selected.indexOf(itemKey) >= 0 &&
+          delete selected[selected.indexOf(itemKey)];
+      else selected = [itemKey];
```

Comment:
This conditional without braces makes the code harder to read at first glance.  It will also be more prone to bugs if someone modifies it later.  Please add braces for both the if and else sections.

Reply:
Handled in 1d4d858